fdbserver/storageserver.actor.cpp

```diff
@@ -10411,9 +10411,28 @@ void changeServerKeysWithPhysicalShards(StorageServer* data,
 			    .detail("PhysicalShard", currentShard->toStorageServerShard().toString());
 		}
 		if (!currentShard.isValid()) {
-			ASSERT(currentRange == keys); // there shouldn't be any nulls except for the range being inserted
+			if (currentRange != keys) {
+				TraceEvent(SevError, "PhysicalShardStateError")
+				    .detail("SubError", "RangeDifferent")
+				    .detail("CurrentRange", currentRange)
+				    .detail("ModifiedRange", keys)
+				    .detail("Assigned", nowAssigned)
+				    .detail("DataMoveId", dataMoveId)
+				    .detail("Version", version);
+				throw internal_error();
+			}
 		} else if (currentShard->notAssigned()) {
-			ASSERT(nowAssigned); // Adding a new range to the server.
+			if (!nowAssigned) {
+				TraceEvent(SevError, "PhysicalShardStateError")
+				    .detail("SubError", "UnassignEmptyRange")
```

Comment:
May want to print detail("CurrentRange", currentRange) here

Reply:
The range information could be found from the log above 'ChangeServerKeysWithPhysicalShards'.